### PR TITLE
CLOUD-2192 update quickstarts for JDV6.4

### PR DESCRIPTION
**Pull request by bdecoste**

### CODE REVIEW

File: datavirt64/dynamicvdb-datafederation/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    ```
+
+    Set this to one of the values listed in the _Datasource Configuration_ section.  Follow the type specific instructions for initializing the database (e.g. seeding the data).  After modifying the deployment configuration, the current deployment should be replaced with a new deployment containing your changes.
+
+> Note: Steps two and three are unnecessary if using the [datavirt-app-secret](https://github.com/jboss-openshift/application-templates/blob/master/secrets/datavirt-app-secret.yaml), as `datavirt-app-config` secret is defined and linked to the service account when that file is installed (e.g. `oc create -f application-templates/secrets/datavirt-app-secret.yaml`).
+
+### Using Red Hat JBoss Data Grid for Materialization
+
+As a prerequisite, you will need to create a data grid service in your OpenShift project.  This can
+be done by using one of the datagrid65-* templates (e.g. datagrid65-basic).  When processing the template,
```
Comment (maschmid) [inline datavirt64/dynamicvdb-datafederation/README.md]: this should recommend using datagrid71 instead

File: datavirt64/dynamicvdb-datafederation/app/datagrid-materialization/pom.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+    <properties>
+
+        <!-- Where to put the EAP module. -->
+        <module.output.dir>${basedir}/../modules/com/client/quickstart/addressbook/pojos/main</module.output.dir>
+
+        <!-- Defines the main class to be execute when running java -jar command -->
+        <main.class.remote-query>org.jboss.as.quickstarts.datagrid.hotrod.query.AddressBookManager</main.class.remote-query>
+
+        <version.org.infinispan.6>6.4.0.Final-redhat-4</version.org.infinispan.6>
```
Comment (maschmid) [inline datavirt64/dynamicvdb-datafederation/app/datagrid-materialization/pom.xml]: should be using a JDG 7.1.1 version instead (which would be 8.4.something I guess)

File: datavirt64/dynamicvdb-datafederation/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+The configuration details are passed to the pod through a secret.
+
+## Deploying the Quickstart
+
+This quickstart can be deployed as follows:
+
+1.  Instantiate one of the following templates:
+
+    * `datavirt63-basic-s2i`
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/README.md]: datavirt64

File: datavirt64/dynamicvdb-datafederation/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    * `datavirt63-secured-s2i` (includes configuration for https and jdbc (over SNI) routes)
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/README.md]: datavirt64 ?

File: datavirt64/dynamicvdb-datafederation/derby/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+##Initializing the Database
+
+No extra steps are required to initialize the database as it is injected into the image during the s2i build (from [../app/data/databases/derby](../app/data/databases/derby)).
+
+##Running the QuickStart
+
+The Derby quickstart requires the Derby JDBC driver to be installed in the JDV image.  This is accomplished by creating a image that contains the driver, as well as an installation script, which adds the driver configuration to the JDV configuration.  An example of a Derby driver image can be found [here](../../derby-driver-image).
+
+The `datavirt63-extensions-support-s2i` template provides support for building the driver extension and integrating it into the JDV s2i process.  This template should be instantiated with the following parameter settings:
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/derby/README.md]: datavirt64 ?

File: datavirt64/dynamicvdb-datafederation/app/pom.xml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.jboss.teiid.quickstart</groupId>
+    <artifactId>datavirt-dynamicvdb-parent</artifactId>
+    <version>3.0.0</version>
+    <packaging>pom</packaging>
+    <name>JDV dyanmicvdb Quickstart Parent</name>
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/app/pom.xml]: dyanmicvdb -> dynamicvdb

File: datavirt64/dynamicvdb-datafederation/app/portfolio-vdb.xml [hunk 7]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<vdb name="Portfolio" version="1">
+
+    <description>The Portfolio Dynamic VDB</description>
+    
+    <!-- 
+      Setting to use connector supplied metadata. Can be "true" or "cached".  
+      "true" will obtain metadata once for every launch of Teiid. 
+      "cached" will save a file containing the metadata into 
+      the deploy/<vdb name>/<vdb version/META-INF directory
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/app/portfolio-vdb.xml]: <vdb version -> <vdb version>

File: datavirt64/hibernate-on-top-of-teiid/README.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+The page should display a list of products.
+
+To add (Register) a new product, enter the following:
+
+-  Product ID  (must be unique)
+-  Company Name
+-  Stock Symbol
+
+> NOTE: the Stock Symbol entered must exist in the marketdata-price.txt file.  The file can be found at ../dynamicvdb-datafederation/src/teiidfiles/data/marketdata-price.txt
```
Comment (penehyba) [inline datavirt64/hibernate-on-top-of-teiid/README.md]: /dynamicvdb-datafederation/**src**/teiidfiles/data/marketdata-price.txt no longer exist. There should be /dynamicvdb-datafederation/**app/data**/teiidfiles/data/marketdata-price.txt

File: datavirt64/dynamicvdb-datafederation/mysql/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+* MYSQL\_USER=myuser
+* MYSQL\_PASSWORD=mypass
+* MYSQL\_DATABASE=accounts
+
+The username and password parameters match what is defined in [datasources.env](../datasources.env).  If you use a different user and/or password, you must modify the settings in this file to match what you're using for the DB.  After modifying the file, you will have to recreate the secret so the application can pick up the changes.  You may also have to trigger a new deployment or recreate the pod, as changes to secrets are not propagated to running pods.
+
+Once the database is up and running, the tables and seed data can be installed by invoking the following:
+
+```
+$ cat customer-schema.sql |  oc exec accounts-mysql5-1-22hwq -i -- scl enable rh-mysql56 -- mysql -u '$MYSQL_USER' '-p$MYSQL_PASSWORD' -h '$ACCOUNTS_MYSQL5_SERVICE_HOST' accounts
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/mysql/README.md]: **'-p**$MYSQL_PASSWORD' -> **-p '**$MYSQL_PASSWORD'

File: datavirt64/dynamicvdb-datafederation/mysql/customer-schema.sql [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+
+CREATE TABLE ACCOUNT
+(
+   ACCOUNT_ID integer,
+   SSN char(10),
+   STATUS char(10),
+   TYPE char(10),
+   DATEOPENED timestamp,
+   DATECLOSED timestamp,
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/mysql/customer-schema.sql]: It's just hunch. But according to setting of **sql_mode**, there could arise error:
ERROR 1067 (42000) at line 57: Invalid default value for 'DATECLOSED'
This is caused by "NO_ZERO_IN_DATE,NO_ZERO_DATE"
mysql> show variables like 'sql_mode' ;
The setting is default in newer MySQL... https://stackoverflow.com/questions/36882149/error-1067-42000-invalid-default-value-for-created-at

File: datavirt64/dynamicvdb-datafederation/README.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+    Specifying parameters as follows:
+
+    * `SOURCE_REPOSITORY_URL`: https://github.com/jboss-openshift/openshift-quickstarts
+    * `SOURCE_REPOSITORY_REF`: master
+    * `CONTEXT_DIR`: datavirt/dynamicvdb-datafederation/app
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/README.md]: datavirt/dynamicvdb-datafederation/app OR datavirt**64**/dynamicvdb-datafederation/app  ?

File: datavirt64/dynamicvdb-datafederation/README.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+### Using Red Hat JBoss Data Grid for Materialization
+
+As a prerequisite, you will need to create a data grid service in your OpenShift project.  This can
+be done by using one of the datagrid71-* templates (e.g. datagrid71-basic).  When processing the template,
+ensure the following parameters are set appropriately:
+
+* `DATAVIRT_CACHE_NAMES`=addressbook
+
+The above will configure three caches: addressbook, addressbook_staging and addressbook_alias.
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/README.md]: CLOUD-2136 will change this to addressbook, ST_addressbook and teiid-alias-naming-cache

File: datavirt64/dynamicvdb-datafederation/mysql/README.md (hunk 9 above)
Comment (penehyba) [reply datavirt64/dynamicvdb-datafederation/mysql/README.md]: in MySQL (Ephemeral) template there is only new version /opt/rh/**rh-mysql57** and this version causes:
ERROR 1067 (42000) at line 57: Invalid default value for 'DATECLOSED'

File: datavirt64/dynamicvdb-datafederation/app/portfolio-vdb.xml [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+    <model name="Accounts">
+        <!-- 
+          JDBC Import settings 
+          
+          importer.useFullSchemaName directs the importer to drop the source 
+          schema from the Teiid object name, so that the Teiid fully qualified name
+          will be in the form of <model name>.<table name>
+        -->
+        <property name="importer.useFullSchemaName" value="false"/>
+
```
Comment (penehyba) [inline datavirt64/dynamicvdb-datafederation/app/portfolio-vdb.xml]: + 
 
        <property name="importer.widenUnsignedTypes" value="false"/>

**Comment by rcernich** [comment]
superseded by #72

**Pull request closed without merging**